src/client/app/reducers/snippet.ts

```diff
@@ -74,6 +74,11 @@ export function reducer(state = initialState, action: SnippetActions | GitHubAct
             return { ...state, running: true };
         }
 
+        case SnippetActionTypes.RUN_FAILED: {
+            AI.trackEvent(action.type);
```

Comment:
Actually, maybe I see now that I look at it again.  Is the idea here that it's more of a CancelRunAction?  If so, maybe rename it to that?  "Failed" implies to me like a message should be shown, but you already show the message at the caller, and this isn't so much a failure as much as a graceful handling of the fact that the run can't physically run.